openlibrary/plugins/openlibrary/js/edit.js

```diff
@@ -146,8 +125,12 @@ function validateIsbn10(data, dataConfig, label) {
     if (isFormatValidIsbn10(data.value) === false) {
         return error('#id-errors', 'id-value', dataConfig['ID must be exactly 10 characters [0-9] or X.'].replace(/ID/, label));
     }
-    else if (isIsbnDupe(data.value) === true) {
-        return error('#id-errors', 'id-value', dataConfig['That ISBN already exists for this edition.'].replace(/ISBN/, label));
+    // hacky special case, dupe checking needs to be here before isbnConfirmAdd
+    // if placed after, duped isbns can be entered by selecting 'Yes' from isbnConfirmAdd
+    // because the isbnConfirmAdd error text with yes/no options is overwritten by dupe error message
+    const entries = document.querySelectorAll(`.${data.name}`);
+    if (isIdDupe(entries, data.value) === true) {
+        return error('#id-errors', 'id-value', dataConfig['That ID already exists for this edition.'].replace(/ID/, label));
```

Comment:
```suggestion
```
We may be able to sort this in the dupe check in `validateIdentifiers()`. See below.